File: analogagenda.client/src/app/DTOs/film.dto.ts
import { FilmType, UsernameType } from "../enums";

export interface FilmDto {
  id: string;
  name?: string;
  brand: string;
  iso: string;
  type: FilmType;
  numberOfExposures: number;
  cost: number;
  costCurrency?: string;
  purchasedBy: UsernameType;
  purchasedOn: string;
  imageUrl: string;
  description: string;
  developed: boolean;
  developedInSessionId?: string | null;
  developedWithDevKitId?: string | null;
  formattedExposureDate?: string;
  photoCount?: number;
  /** UTC from API; appended as `UpdatedDate` on film image display URLs. */
  updatedDate?: string | null;
}

File: Database/DTOs/FilmDto.cs
namespace Database.DTOs;

public class FilmDto
{
    public string Id { get; set; } = string.Empty;

    public string Name { get; set; } = string.Empty;

    public required string Brand { get; set; }

    public required string Iso { get; set; }

    public required string Type { get; set; }

    public int NumberOfExposures { get; set; } = 36;

    public double Cost { get; set; }

    public required string PurchasedBy { get; set; }

    public DateOnly PurchasedOn { get; set; }

    public string ImageUrl { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public bool Developed { get; set; }

    public string? DevelopedInSessionId { get; set; }

    public string? DevelopedWithDevKitId { get; set; }

    public string FormattedExposureDate { get; set; } = string.Empty;

    public int PhotoCount { get; set; } = 0;
}
